### PR TITLE
Add support to calculate npm tree only by package-lock.json file

### PR DESCRIPTION
**Pull request by asafambar**
- [ ] All [tests](https://github.com/jfrog/build-info-go#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [ ] All [static analysis checks](https://github.com/jfrog/build-info-go/actions/workflows/analysis.yml) passed.
- [ ] This pull request is on the dev branch.
- [ ] I used gofmt for formatting the code before submitting the pull request.
-----

This PR introduce more control on the possible ways npm tree is calculated:
1) create the tree (which is the "ls" command) only by the package-lock file, with no mater if node modules exists.
2) if package-lock exists, check if the package.json is more updated than the lock, if so, then recalculate th package-lock file.

### CODE REVIEW

**Comment by asafambar** [comment]
This PR introduce more control on the possible ways npm tree is calculated:
1) create the tree (which is the "ls" command) only b the package-lock file, with no mater if noe modules exists.
2) if package-lock exists, check if the package.json is more updated than the lock, if so, then recalculate th package-lock file.

File: build/utils/npm_test.go [hunk 1]
```diff
@@ -192,12 +195,92 @@ func TestDependencyWithNoIntegrity(t *testing.T) {
...
+	if !npmVersion.AtLeast("7.0.0") {
+		t.Skip("Running on npm v7 and above only, skipping...")
+	}
+	path, cleanup := testdatautils.CreateTestProject(t, filepath.Join("..", "testdata/npm/project6"))
+	defer cleanup()
+	data, err := os.ReadFile(filepath.Join(path, "package-lock_test.json"))
+	require.NoError(t, err)
+	info, err := os.Stat(filepath.Join(path, "package-lock_test.json"))
+	require.NoError(t, err)
+	require.NoError(t, os.WriteFile(filepath.Join(path, "package-lock.json"), data, info.Mode().Perm()))
```
Comment (yahavi) [inline build/utils/npm_test.go:218]: Can be one lined:
```suggestion
	assert.NoError(t, utils.MoveFile(filepath.Join(path, "package-lock_test.json"), filepath.Join(path, "package-lock.json")))
```

File: build/utils/npm_test.go [hunk 2]
```diff
@@ -192,12 +195,92 @@ func TestDependencyWithNoIntegrity(t *testing.T) {
...
+	// sleep so the package.json modified time will be bigger than the package-lock.json, this make sure it will recalculate lock file.
+	time.Sleep(time.Millisecond * 20)
+	require.NoError(t, os.Chtimes(filepath.Join(path, "package.json"), time.Now(), time.Now()))
```
Comment (yahavi) [inline build/utils/npm_test.go:221]: Can be without sleep
```suggestion
	require.NoError(t, os.Chtimes(filepath.Join(path, "package.json"), time.Now(), time.Now().Add(time.Millisecond * 20)))
```

File: build/utils/npm_test.go [hunk 3]
```diff
@@ -192,12 +195,92 @@ func TestDependencyWithNoIntegrity(t *testing.T) {
...
+				RequestedBy: [][]string{{"jfrogtest"}},
+				Checksum:    entities.Checksum{},
+			},
+			npmLsDependency: &npmLsDependency{
+				Name:      "minimist",
+				Version:   "0.1.0",
+				Integrity: "sha512-wR5Ipl99t0mTGwLjQJnBjrP/O7zBbLZqvA3aw32DmLx+nXHfWctUjzDjnDx09pX1Po86WFQazF9xUzfMea3Cnw==",
+			},
+		},
+	}
```
Comment (yahavi) [inline build/utils/npm_test.go:280]: For better test readability, is it possible to export this to a const outside of the test method?

**Review by yahavi — CHANGES_REQUESTED** [review]

**Review by yahavi — APPROVED** [review]

**Pull request merged**